lib/route/src/Obelisk/Route/Frontend.hs

```diff
@@ -519,27 +521,55 @@ routeLink
   -> m a -- ^ Child widget
   -> m a
 routeLink r w = do
-  (e, a) <- routeLinkImpl r w
+  (e, a) <- routeLinkImpl mempty r w
   scrollToTop e
   return a
 
+-- | Like 'routeLink', but takes additional attributes as argument.
+--
+routeLinkAttr
+  :: forall t m a route js.
+     ( DomBuilder t m
+     , RouteToUrl route m
+     , SetRoute t route m
+     , Prerender js t m
+     )
+  => Map AttributeName Text -- ^ Additional attributes
+  -> route -- ^ Target route
+  -> m a -- ^ Child widget
+  -> m a
+routeLinkAttr attrs r w = do
+  (e, a) <- routeLinkImpl attrs r w
+  let
+    targetBlank = Map.lookup "target" attrs == Just "_blank"
+  when (not targetBlank) $ scrollToTop e
+  return a
+
 -- | Raw implementation of 'routeLink'. Does not scroll to the top of the page on clicks.
 routeLinkImpl
   :: forall t m a route.
      ( DomBuilder t m
      , RouteToUrl route m
      , SetRoute t route m
      )
-  => route -- ^ Target route
+  => Map AttributeName Text
+  -> route -- ^ Target route
   -> m a -- ^ Child widget
   -> m (Event t (), a)
-routeLinkImpl r w = do
+routeLinkImpl attrs r w = do
   enc <- askRouteToUrl
-  let cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
-        & elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault)
-        & elementConfig_initialAttributes .~ "href" =: enc r
+  let
+    -- If targetBlank == True, the link will be opened in another page. In that
+    -- case, we don't prevent the default behaviour, and we don't need to
+    -- setRoute.
+    targetBlank = Map.lookup "target" attrs == Just "_blank"
+    cfg = (def :: ElementConfig EventResult t (DomBuilderSpace m))
+        & elementConfig_initialAttributes .~ ("href" =: enc r <> attrs)
+        & (if targetBlank
+           then id
+           else elementConfig_eventSpec %~ addEventSpecFlags (Proxy :: Proxy (DomBuilderSpace m)) Click (\_ -> preventDefault))
   (e, a) <- element "a" cfg w
-  setRoute $ r <$ domEvent Click e
+  when (not targetBlank) $ setRoute $ r <$ domEvent Click e
```

Comment:
Is there a way to factor out the `targetBlank` check so we only have to do it once? Perhaps `scrollToTop` could be moved into `routeLinkImpl`?

Reply:
`routeLinkImpl` was used for the no-scroll-to-top version too. That function seems to have gone though, so I'm fine with moving the scrollToTop call into here. Otherwise we can just return `targetBlank` so we don't duplicate the check - this function isn't exported.

Reply:
We also need to do this check for the dynamic attr version.